assignments/array-methods.js

```diff
@@ -56,28 +56,61 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(x=>{
+     fullName.push(`${x.first_name} ${x.last_name}`)
+})
 console.log(fullName);
 
+
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+let allCaps = runners.map(e=>{
+    return`${e.first_name.toUpperCase()}`
+});
+
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter(x=>{
+   return (x.shirt_size==='L')
+});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+ticketPriceTotal.push( runners.reduce((acc,cur,idx,src)=>{
+    return acc+cur.donation;
+},0))
 console.log(ticketPriceTotal);
-
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
```

Comment:
This is a great opportunity to practice using helpful comments. Your comments here could have told me, or any future user, what problems you decided to solve. This is a valuable skill as it makes your code readable for future developers (and for any manager/supervisor/Team Lead whose job it is to review your code).

Reply:
Thank you for the criticism. Now that you mention it I would have been ticked if I had run across this myself in the wild.